realm/realm-library/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -967,7 +967,7 @@ public String toString() {
                     RealmFieldType.OBJECT.name(), RealmFieldType.LIST.name()));
         }
 
-        return RealmResults.createBacklinkResults(realm, (CheckedRow) proxyState.getRow$realm(), realmObjectSchema.getTable(), srcFieldName);
+        return RealmResults.createDynamicBacklinkResults(realm, proxyState.getRow$realm(), realmObjectSchema.getTable(), srcFieldName);
```

Comment:
Now that you mention it, separating the call based on a cast to `(UncheckedRow)` OR `(CheckedRow)` **is** super brittle.